### PR TITLE
Only prompt if input

### PR DESCRIPTION
**Pull request by meganrogge**
This PR intends to only show the prompt if the terminal instance that's active has had keyboard input. It does not deal with the case of typing and deleting all that you typed to simplify the code.

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts [hunk 1]
```diff
@@ -544,11 +544,19 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				return false;
 			}
 
+			// only keep track of input if prompt hasn't already been shown
+			if (this._storageService.getBoolean(SHOW_TERMINAL_CONFIG_PROMPT, StorageScope.GLOBAL, true) &&
+				// just a normal key with no modifier
+				event.key.match(/^[a-z0-9]$/i) && !event.ctrlKey && !event.shiftKey && !event.altKey) {
+				this._storageService.store(HAS_TYPED_IN_TERMINAL + this.id, true, StorageScope.GLOBAL, StorageTarget.USER);
```
Comment (roblourens) [inline src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:551]: Does the storage service need to be involved?

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts [hunk 2]
```diff
@@ -544,11 +544,19 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				return false;
 			}
 
+			// only keep track of input if prompt hasn't already been shown
+			if (this._storageService.getBoolean(SHOW_TERMINAL_CONFIG_PROMPT, StorageScope.GLOBAL, true) &&
+				// just a normal key with no modifier
+				event.key.match(/^[a-z0-9]$/i) && !event.ctrlKey && !event.shiftKey && !event.altKey) {
```
Comment (roblourens) [inline src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:550]: I'm not sure whether this regex is safe for non-english languages...

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:550]: my guess is no. not sure how to address this

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts (hunk 1 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:551]: probably not. I could make the TerminalInstance have an `hasHadInput` boolean

File: src/vs/workbench/contrib/terminal/common/terminal.ts [hunk 3]
```diff
@@ -51,7 +51,11 @@ export const KEYBINDING_CONTEXT_TERMINAL_PROCESS_SUPPORTED = new RawContextKey<b
 export const IS_WORKSPACE_SHELL_ALLOWED_STORAGE_KEY = 'terminal.integrated.isWorkspaceShellAllowed';
 export const NEVER_MEASURE_RENDER_TIME_STORAGE_KEY = 'terminal.integrated.neverMeasureRenderTime';
 export const SHOW_TERMINAL_CONFIG_PROMPT = 'terminal.integrated.showTerminalConfigPrompt';
-export const HAS_TYPED_IN_TERMINAL = 'terminal.integrated.hasTypedInput';
+
+export const EXCLUDED_KEYS = ['RightArrow', 'LeftArrow', 'UpArrow', 'DownArrow', 'Space', 'Meta', 'Control', 'Shift', 'Alt', '', 'Delete', 'Backspace', 'Tab'];
```
Comment (roblourens) [inline src/vs/workbench/contrib/terminal/common/terminal.ts:55]: I would move these two consts to be closer to the code that uses them. They don't seem like generic terminal concepts to me. Like it's not clear what "excluded" without knowing the context. Everything else looks good!

**Pull request merged**